clients/src/test/java/org/apache/kafka/clients/producer/internals/ProducerInterceptorsTest.java

```diff
@@ -126,7 +126,7 @@ public void onAcknowledgement(RecordMetadata metadata, Exception exception, Head
                 onErrorAckCount++;
                 // the length check is just to call topic() method and let it throw an exception
                 // if RecordMetadata.TopicPartition is null
-                if (metadata != null && metadata.topic().length() >= 0) {
+                if (metadata != null && !metadata.topic().isEmpty()) {
```

Comment:
i don't think this is the same thing, the old check was `>=` not just `>` -- if anything it seems like we should remove the entire part of the conditional because isn't `length() >= 0` trivially true? what is the topic, a string?

Reply:
oh nvm, didn't read the comment above 🤦‍♀️

Reply:
kind of worried someone else might see this and miss the comment above like I did and end up removing what looks like a no-op check. can we take the `metadata.topic()` out of the conditional to make it more obvious?

Reply:
Yes, `topic()` returns a string. Cf the comment though -- it's really just about calling `topic()` to begin with...

Reply:
better yet just this:

```
 if (metadata != null) {
     if (metadata.topic() == null) {
        throw new WhateverException();

```

Reply:
Yeah. Good idea. Let's do it this way.